pilot/pkg/networking/plugin/plugin.go

```diff
@@ -47,7 +50,29 @@ const (
 )
 
 // ModelProtocolToListenerProtocol converts from a config.Protocol to its corresponding plugin.ListenerProtocol
-func ModelProtocolToListenerProtocol(protocol config.Protocol) ListenerProtocol {
+func ModelProtocolToListenerProtocol(node *model.Proxy, protocol config.Protocol) ListenerProtocol {
+	switch protocol {
+	case config.ProtocolHTTP, config.ProtocolHTTP2, config.ProtocolGRPC, config.ProtocolGRPCWeb:
+		return ListenerProtocolHTTP
+	case config.ProtocolTCP, config.ProtocolHTTPS, config.ProtocolTLS,
+		config.ProtocolMongo, config.ProtocolRedis, config.ProtocolMySQL:
+		return ListenerProtocolTCP
+	case config.ProtocolUnsupported:
+		if util.IsProxyVersionGE13(node) {
+			return ListenerProtocolAuto
+		}
+
+		return ListenerProtocolUnknown
+	default:
+		return ListenerProtocolUnknown
+	}
+}
+
+func DefaultListenerProtocol(node *model.Proxy, protocol config.Protocol) ListenerProtocol {
+	if util.IsProxyVersionGE13(node) {
```

Comment:
I'm not sure if the e2e test environment is using a proxy with version >= 1.3. I was under the impression it's not updated for 1.3 yet which means this new code path may not be tested by the e2e tests.